### DISCUSSION TITLE
Pycharm is not recognizing return type for select & scalar statements

### DISCUSSION THREAD

**Discussion by sombek** [sqlalchemy/sqlalchemy#9533] [Usage Questions]
Hello

I am trying to utilize the new version of sqlalchemy typing.
I was following this examples:
https://docs.sqlalchemy.org/en/20/changelog/whatsnew_20.html#migrating-an-existing-mapping

I created User table
```python
    class Base(DeclarativeBase):
        pass

    class User(Base):
        __tablename__ = "user_account"

        id: Mapped[int] = mapped_column(primary_key=True)
        name: Mapped[str]
```

Then when I try to use it in select statement:
![image](https://user-images.githubusercontent.com/20394589/227508404-5d37858c-dd09-4175-8943-6b3878d3b533.png)
I am getting [Any, Any] not what I expected as per the documentation.

But when I select the whole User class I am getting what I expect:
![image](https://user-images.githubusercontent.com/20394589/227508818-488e79c8-b74e-40c9-8794-a31097d75940.png)

The other issue is when I try to run 
```python
session.execute(stmt_2).scalars().all()
```
I am not getting the correct type
![image](https://user-images.githubusercontent.com/20394589/227508900-d2723a48-1578-479b-8a5b-a636ecf02a75.png)

I am not using any sqlalchemy stubs. And `mypy.ini` plugins tag is empty 

My setup is:
mypy==1.1.1
sqlalchemy==2.0.7

Using python 3.11.2
Pycharm as IDE

**Comment by CaselIT** [comment]
Hi,

The tooltip is of pychartm not of mypy.
This is similar to the other issues with pycharm.

There isn't much that sqlalchemy can do here, you need to open an issue with pycharm

**Reply by CaselIT** [reply]
I've created a label for it, you can now search them https://github.com/sqlalchemy/sqlalchemy/discussions?discussions_q=label%3Apycharm+
